### PR TITLE
[LU 177] Recruiting OpenAPI

### PR DESCRIPTION
**Pull request by clehel**

### CODE REVIEW

File: personio-recruiting-api.yaml [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+              schema:
+                $ref: "#/components/schemas/XmlResponse"
+  /recruiting/applicant:
+    post:
+      security:
+        - BearerAuth: [ ]
+      tags:
+        - Applicant
+      summary: Passing applications to Personio
+      description: Description of the applicant endpoint of the Personio Recruiting API
```
[the remark this answers is not in the document]
Comment (clehel) [reply personio-recruiting-api.yaml:64]: We're moving info from https://developer.personio.de/docs/applications-endpoint - we were told this is not a priority and will be sorted out later

File: personio-recruiting-api.yaml [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+              type: string
+            value:
+              type: string
+
+    Applicant:
+      title: Applicant
+      type: object
+      properties:
+        company_id:
+          description: Your company ID (see above)
```
[the remark this answers is not in the document]
Comment (clehel) [reply personio-recruiting-api.yaml]: Thanks, corrected: referred user to endpoint description.

File: personio-recruiting-api.yaml [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+          type: integer
+        access_token:
+          description: API Access token for your company account (see above)
```
[the remark this answers is not in the document]
Comment (clehel) [reply personio-recruiting-api.yaml]: Thanks, corrected: referred user to endpoint description.

File: personio-recruiting-api.yaml [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+          type: object
+          allOf:
+            - $ref: "#/components/schemas/Document"
+        message:
+          description: Initial message from the applicant
+          type: string
+        tags:
+          description: Existing tags (new ones cannot be created via API) <br/>
+            [array of strings], e.g. [ “tag_1”, “tag_2" ]
+        birthday:
```
[the remark this answers is not in the document]
Comment (clehel) [reply personio-recruiting-api.yaml:382]: Thanks, corrected.

File: personio-recruiting-api.yaml [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+          maxLength: 255
+        location:
+          description: Current location of the applicant
+          type: string
+          maxLength: 255
+        salary_expectations:
+          description: Salary expectations of the applicant (Will not be parsed, so you can transfer values like "minimum 50k")
+          type: string
+          maxLength: 255
+        available_from:
```
[the remark this answers is not in the document]
Comment (clehel) [reply personio-recruiting-api.yaml:338]: Information was taken from existing documentation, which only states `string (max:100)`

File: personio-recruiting-api.yaml [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+          type: string
+          maxLength: 255
+        gender:
+          description: Gender of the applicant
+          type: string
+          enum: [ male, female, diverse ]
+        recruiting_channel_id:
+          description: ID of the recruiting channel that this applicant applied through recruiting_channel_id has to match the id of a channel you created in Personio
+          type: integer
+        phone:
```
[the remark this answers is not in the document]
Comment (clehel) [reply personio-recruiting-api.yaml:326]: Information was taken from existing documentation, which only states `string (max:255)`

File: personio-recruiting-api.yaml [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+            [array of files] <br/>
+            allowed extensions are 'pdf', 'docx', 'doc', 'jpg', 'png', 'txt', 'jpeg', 'odt', 'ods','xlsx', 'rtf', 'xls', 'pptx', 'ppt', 'gif', 'tif', 'tiff', 'bmp', 'csv', 'rar', 'gz', 'zip', '7z'; <br/>
+            filesize per document < 20M; <br/>
+            total post size < 64M; <br/>
+            [array of category names] <br/>
+            allowed values are 'cv', 'cover-letter', 'employment-reference', 'certificate', 'work-sample' or 'other' <br/>
+          type: array
+          items:
+            $ref: "#/components/schemas/Document"
+        documentn:
```
[the remark this answers is not in the document]
Comment (clehel) [reply personio-recruiting-api.yaml]: Thanks - not a typo, but indeed needs a better name - changed it to `document{n}`

File: personio-recruiting-api.yaml [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+          type: string
+          enum: [ male, female, diverse ]
+        recruiting_channel_id:
+          description: ID of the recruiting channel that this applicant applied through recruiting_channel_id has to match the id of a channel you created in Personio
+          type: integer
+        phone:
+          description: Phone number of the applicant
+          type: string
+          maxLength: 255
+        location:
```
[the remark this answers is not in the document]
Comment (clehel) [reply personio-recruiting-api.yaml:330]: Information was taken from existing documentation, which only states string `(max:255)` - will try to get more info regarding this property

**Review by iFoukis — APPROVED** [review]

**Pull request merged**